lib/accessors/float.js

```diff
@@ -3,7 +3,7 @@
 module.exports = function asFloat (value) {
   const n = parseFloat(value)
 
-  if (isNaN(n) || n.toString() !== value) {
+  if (isNaN(n) || isNaN(value)) {
```

Comment:
Nice find on this bug, and neat solution. I had to experiment with `isNaN` to understand it.

Can you add a comment explaining why this second isNaN check is performed?